File: linear_search/linear-search.go
package main

import "fmt"

func linearSearch(array []int, needle int) int {
	for i, item := range array {
		if item == needle {
			return i
		}
	}
	return -1
}

func main() {
	array := []int{0, 2, 4, 6, 8, 10, 12, 14, 16, 18}
	index := linearSearch(array, 10)
	if index == -1 {
		fmt.Println("Number not found")
	} else {
		fmt.Println("10 found at index", index)
	}
}

File: linear_search/LinearSearch.dart
int LinearSearch(List<int> a, number) {
  for (int i = 0; i < a.length; i++) {
    if (a[i] == number) {
      return i;
    }
  }
  return -1;
}

void main() {
  List<int> list = [10,20,30,40,50,60,70,80,90,100];
  int x = 30;
  int index = LinearSearch(list, x);
  
  if (index != -1) {
    print('$x found at positions: $index');
  } else {
    print('$x Not found');
  }
}